doc/source/examples/incompressible-flow/2d-sudden-expansion-flow/2d-sudden-expansion-flow.rst

```diff
@@ -0,0 +1,332 @@
+      set method                       = steady
+      set number mesh adapt            = 9
+      set output name                  = sudden-expansion-flow-output
+      set output frequency             = 1
+    end
+
+For :math:`\textrm{Re} = 610`, a transient second-order backward differentiation scheme is used:
+
+.. code-block:: text
+
+    subsection simulation control
+      set method           = bdf2
+      set output name      = sudden-expansion-flow-output
+      set output frequency = 15
+      set adapt            = true
+      set max cfl          = 5
+      set time step        = 0.001
+      set time end         = 15
+    end
+
+in which ``adapt``, ``max cfl``, and ``time step`` parameters are defined so that the time step is small enough to capture the flow behaviour without large damping oscillations. The final simulation time is taken as :math:`t_{end} = 15s`.
+
+Physical Properties
+~~~~~~~~~~~~~~~~~~~
+
+The definition of the Reynolds number in this problem is given by
+
+.. math::
+    \mathrm{Re} = \frac{u h}{\nu}
+
+where :math:`\nu` is the kinematic viscosity. Considering that the stream velocity is :math:`u = 1 m/s`, the physical properties are then defined as:
+
+.. code-block:: text
+
+    subsection physical properties
+        set number of fluids = 1
+        subsection fluid 0
+            set kinematic viscosity = 1.42857e-4 # Re_h=u*h/nu
+        end
+    end
+
+The viscosity :math:`\nu = 1.42857 \times 10^{-4} m^2/s` corresponds to :math:`\textrm{Re} = 70`. For :math:`\textrm{Re} = 610`, that should be changed to :math:`\nu = 1.6393 \times 10^{-5} m^2/s`.
+
+Mesh
+~~~~
+
+The mesh is composed by bilinear quadrilateral elements, and the problem geometry was taken from Durst et. al. (1993) [#durst1993]_. The inlet dimension was chosen to be large enough to for the flow to fully develop before the channel expansion.
+
+.. code-block:: text
+
+    subsection mesh
+      set type      = gmsh
+      set file name = ./sudden-expansion-flow.msh
+    end
+
+Mesh Adaptation
+~~~~~~~~~~~~~~~
+
+The mesh adaptation algorithm is based on the Kelly error estimator applied on the velocity variable. For :math:`\textrm{Re} = 70` the correspondent code section looks as follows:
+
+.. code-block:: text
+
+    subsection mesh adaptation
+      set variable             = velocity
+      set type                 = kelly
+      set fraction refinement  = 0.2
+      set fraction coarsening  = 0
+      set frequency            = 1
+      set min refinement level = 0
+      set max refinement level = 8
+    end
+
+In this case, the algorithm thoroughly discretizes the mesh around the expansion section, where the recirculation zones appear:
+
+.. image:: image/mesh-adaptation-Re70.png
+
+For :math:`\textrm{Re} = 610`, mesh adaptation was limited by the maximum refinement level, the fraction refinement, and the number of cells, so that the mesh discretization did not became too computationally expensive at the final simulation time. 
```

Comment:
Be careful, you are using the mesh refinement controller. So this feature actually controls the adaptation rate to ensure we reach a target number of cell. It does so by playing with the coarsening ratio and refinement ratio. This is a nice feature that @Luckabarbeau added.